x-pack/legacy/plugins/uptime/public/components/functional/charts/donut_chart.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React, { useContext, useEffect, useRef } from 'react';
+import * as d3 from 'd3';
+import { DonutChartLegend } from './donut_chart_legend';
+import { UptimeSettingsContext } from '../../../contexts';
+
+interface DonutChartProps {
+  down: number;
+  height: number;
+  up: number;
+  width: number;
+}
+
+export const DonutChart = ({ height, down, up, width }: DonutChartProps) => {
+  const chartElement = useRef<SVGSVGElement | null>(null);
+
+  const {
+    colors: { danger, gray },
+  } = useContext(UptimeSettingsContext);
+
+  useEffect(() => {
+    if (chartElement.current !== null) {
+      const svgElement = d3
+        .select(chartElement.current)
+        .append('g')
+        .attr('transform', `translate(${width / 2}, ${height / 2})`);
+      const color = d3.scale
+        .ordinal()
+        .domain(['up', 'down'])
+        .range([gray, danger]);
+      const pieGenerator = d3.layout
+        .pie()
+        .value(({ value }: any) => value)
+        // these start/end angles will reverse the direction of the pie,
+        // which matches our design
+        .startAngle(2 * Math.PI)
+        .endAngle(0);
+
+      svgElement
+        .selectAll('g')
+        .data(
+          // @ts-ignore pie generator expects param of type number[], but only works with
+          // output of d3.entries, which is like Array<{ key: string, value: number }>
+          pieGenerator(d3.entries({ up, down }))
+        )
+        .enter()
+        .append('path')
+        .attr(
+          'd',
+          // @ts-ignore attr does not expect a param of type Arc<Arc> but it behaves as desired
+          d3.svg
+            .arc()
+            .innerRadius(width * 0.2)
```

Comment:
An innerRadius of `width * 0.28` seems perfect to me.
<img width="308" alt="Screenshot 2019-10-09 at 10 52 42" src="https://user-images.githubusercontent.com/20965076/66466551-30b08180-ea83-11e9-9d86-a97b32ab6b08.png">

Reply:
Fixed in https://github.com/elastic/kibana/pull/47176/commits/8c65f72f90a1af07381c4a7e05044958f4c22214.